### PR TITLE
update vendor

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Notably, github.com/vishvananda/netlink hadn't been updated since 2015.

---
Also run `make verify-dependencies` with Go 1.14.x

Background: https://github.com/opencontainers/runc/pull/2073#issuecomment-596068439

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Travis wasn't triggered?

**Comment by AkihiroSuda** [comment]
failing
```
not ok 75 update
# (in test file tests/integration/update.bats, line 121)
#   `[ "$status" -eq 0 ]' failed
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
# runc spec (status=0):
# 
# runc run -d --console-socket /tmp/console.sock test_update (status=0):
# 
# runc update test_update --cpu-period 900000 (status=0):
# 
# runc update test_update --cpu-quota 600000 (status=0):
# 
# runc update test_update --cpu-share 200 (status=0):
# 
# runc update test_update --cpuset-cpus 1 (status=0):
# 
# runc update test_update --memory 67108864 (status=0):
# 
# runc update test_update --memory 50M (status=0):
# 
# runc update test_update --memory-reservation 33554432 (status=0):
# 
# runc update test_update --memory-swap -1 (status=1):
# Incorrect Usage: flag provided but not defined: -1
# 
# NAME:
#    runc update - update container resource constraints
# 
# USAGE:
#    runc update [command options] <container-id>
# 
# OPTIONS:
#    --resources value, -r value  path to the file containing the resources to update or '-' to read from the standard input
# 
# The accepted format is as follow (unchanged values can be omitted):
# 
# {
#   "memory": {
#     "limit": 0,
#     "reservation": 0,
#     "swap": 0,
#     "kernel": 0,
#     "kernelTCP": 0
#   },
#   "cpu": {
#     "shares": 0,
#     "quota": 0,
#     "period": 0,
#     "realtimeRuntime": 0,
#     "realtimePeriod": 0,
#     "cpus": "",
#     "mems": ""
#   },
#   "blockIO": {
#     "weight": 0
#   }
# }
# 
# Note: if data is to be read from a file or the standard input, all
# other options are ignored.
#    --blkio-weight value        Specifies per cgroup weight, range is from 10 to 1000 (default: 0)
#    --cpu-period value          CPU CFS period to be used for hardcapping (in usecs). 0 to use system default
#    --cpu-quota value           CPU CFS hardcap limit (in usecs). Allowed cpu time in a given period
#    --cpu-share value           CPU shares (relative weight vs. other containers)
#    --cpu-rt-period value       CPU realtime period to be used for hardcapping (in usecs). 0 to use system default
#    --cpu-rt-runtime value      CPU realtime hardcap limit (in usecs). Allowed cpu time in a given period
#    --cpuset-cpus value         CPU(s) to use
#    --cpuset-mems value         Memory node(s) to use
#    --kernel-memory value       Kernel memory limit (in bytes)
#    --kernel-memory-tcp value   Kernel memory limit (in bytes) for tcp buffer
#    --memory value              Memory limit (in bytes)
#    --memory-reservation value  Memory reservation or soft_limit (in bytes)
#    --memory-swap value         Total memory usage (memory + swap); set '-1' to enable unlimited swap
#    --pids-limit value          Maximum number of pids allowed in the container (default: 0)
#    --l3-cache-schema value     The string of Intel RDT/CAT L3 cache schema
#    --mem-bw-schema value       The string of Intel RDT/MBA memory bandwidth schema
#    
# time="2020-03-24T18:53:58Z" level=error msg="flag provided but not defined: -1"
# flag provided but not defined: -1
# runc list (status=0):
# ID            PID         STATUS      BUNDLE             CREATED                          OWNER
# test_update   17740       running     /tmp/busyboxtest   2020-03-24T18:53:57.736578651Z   root
# runc kill test_update KILL (status=0):
# 
# runc delete test_update (status=0):
# 
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
# runc list (status=0):
# ID          PID         STATUS      BUNDLE      CREATED     OWNER
```

**Comment by AkihiroSuda** [comment]
Opened https://github.com/urfave/cli/issues/1092 for reporting the urfave/cli regression in v1.22.2.

**Comment by AkihiroSuda** [comment]
rebased, but travis seems dead 😕

**Comment by kolyshkin** [comment]
From CI
> git status:
> ...
> modified:   vendor/modules.txt

This is from `make verify-dependencies` which Travis runs with go 1.13. Maybe you did it with go 1.14 and there are some differences?

**Comment by AkihiroSuda** [comment]
Seems we should use Go 1.14 for verifying vendor: https://github.com/opencontainers/runc/pull/2073#issuecomment-596068439

> switched back to running the vendoring in 1.14. If we vendor with go 1.13, go build will fail on go 1.14, making the tests fil.

**Comment by iwankgb** [comment]
I think that this PR needs to be merged sooner rather then later as it fixes issue that blocks some other changes (#2271, #2276 and #2278) while #2277 contains an alternative fix.

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ go:
 
 matrix:
   include:
-    - go: 1.13.x
+    - go: 1.x
```
Comment (saschagrunert) [inline .travis.yml]: If go 1.15 comes out which may introduce changes to go modules, then this might silently break CI via `make verify-dependencies`.

Comment (AkihiroSuda) [reply .travis.yml]: good point, fixed

**Comment by mrunalp** [comment]
LGTM

**Comment by mrunalp** [comment]
@opencontainers/runc-maintainers ptal.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**